File: message/src/main/java/com/ricardococati/kafka/MessageExecutorCandleSemanal.java
package com.ricardococati.kafka;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.ricardococati.model.dto.CandlestickSemanalMessage;
import com.ricardococati.service.CandlestickSemanalInserirService;
import java.io.IOException;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
@Slf4j
public class MessageExecutorCandleSemanal {

  private final ObjectMapper objectMapper;
  private final CandlestickSemanalInserirService service;

  public void execute(final String payload) {
    final CandlestickSemanalMessage domain = payloadToDomain(payload);
    log.info("payload: " + payload);
    service.incluirCandlestickSemanal(domain);
  }

  private CandlestickSemanalMessage payloadToDomain(String payload) {
    try {
      return objectMapper.readValue(payload, CandlestickSemanalMessage.class);
    } catch (IOException e) {
      throw new RuntimeException(e);
    }
  }
}

File: message/src/main/kotlin/com/ricardococati/kafka/MessageExecutorCandleSemanal.kt
package com.ricardococati.kafka

import com.fasterxml.jackson.databind.ObjectMapper
import com.ricardococati.model.entities.CandlestickSemanalMessage
import com.ricardococati.service.CandlestickSemanalInserirService
import lombok.RequiredArgsConstructor
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import java.io.IOException

@Service
@RequiredArgsConstructor
class MessageExecutorCandleSemanal {

    private val objectMapper: ObjectMapper? = null
    private val service: CandlestickSemanalInserirService? = null
    fun execute(payload: String) {
        val domain = payloadToDomain(payload)
        log.info("payload: $payload")
        service!!.incluirCandlestickSemanal(domain)
    }

    private fun payloadToDomain(payload: String): CandlestickSemanalMessage {
        return try {
            objectMapper!!.readValue(payload, CandlestickSemanalMessage::class.java)
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }

    companion object {
        private val log = LoggerFactory.getLogger(MessageExecutorCandleSemanal::class.java)
    }

}